File: core/core-api/src/main/java/com/coffee/gu/api/controller/v1/request/CreatePaymentRequest.java
package com.coffee.gu.api.controller.v1.request;

import com.coffee.gu.coupon.IssuedCoupon;
import com.coffee.gu.payment.PaymentDiscount;

import java.math.BigDecimal;
import java.util.List;

public record CreatePaymentRequest(
        String orderKey,
        Long usedIssuedCouponId
) {
    public PaymentDiscount toPaymentDiscount(List<IssuedCoupon> issuedCoupons, BigDecimal orderAmount) {
        return PaymentDiscount.of(issuedCoupons, usedIssuedCouponId, orderAmount);
    }
}

File: core/core-api/src/main/kotlin/com/coffee/gu/api/controller/v1/request/CreatePaymentRequest.kt
package com.coffee.gu.api.controller.v1.request

import com.coffee.gu.coupon.IssuedCoupon
import com.coffee.gu.payment.PaymentDiscount
import java.math.BigDecimal

class CreatePaymentRequest(
    val orderKey: String,
    val usedIssuedCouponId: Long?,
) {
    fun toPaymentDiscount(issuedCoupons: List<IssuedCoupon>, orderAmount: BigDecimal): PaymentDiscount {
        return PaymentDiscount.of(issuedCoupons, usedIssuedCouponId, orderAmount)
    }
}
